### ISSUE TITLE
Cannot install Kibana: required "node" v8.11.4 cannot be compiled with GCC 5.5.0

### ISSUE THREAD

**Issue by StudioMaX** [Linuxbrew/homebrew-core#9530]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single formula? If it's a general `brew` problem please file this issue at Linuxbrew/brew: https://github.com/Linuxbrew/brew/issues/new/choose. If it's a tap (e.g. Brewsci/homebrew-bio) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link? https://gist.github.com/StudioMaX/6c309324e57a65484af54ae08e185801
- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)
Install Kibana on Ubuntu 16.04. It requires and tries to compile `node` (as I understood, not as a brew dependency, just as additional source package), but `node` is already installed via `brew install node`.

- What happened (include command output)
```
$ brew install kibana
==> Cloning https://github.com/elastic/kibana.git
Cloning into '/usr/home/lynpub/.cache/Homebrew/kibana--git'...
Note: checking out '2872c74872c54d91e3e00b67c7bbc61659df0ba7'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

==> Checking out tag v6.4.1
HEAD is now at 2872c74... eui 3.0.6, fix for ie11 tooltip (#23006)
==> Downloading https://nodejs.org/dist/v8.11.4/node-v8.11.4.tar.xz
######################################################################## 100.0%
==> ./configure --prefix=/usr/home/lynpub/.linuxbrew/Cellar/kibana/6.4.1/libexec/node
==> make install
Last 15 lines from /usr/home/lynpub/.cache/Homebrew/Logs/kibana/02.make:
   if (raw_keylen < 0.0 || std::isnan(raw_keylen) || std::isinf(raw_keylen) ||
                           ^
../src/node_crypto.cc:5519:27: note: suggested alternative:
<built-in>: note:   '__builtin_isnan'
../src/node_crypto.cc:5519:53: error: '__builtin_isinf_sign' is not a member of 'std'
   if (raw_keylen < 0.0 || std::isnan(raw_keylen) || std::isinf(raw_keylen) ||
                                                     ^
../src/node_crypto.cc:5519:53: note: suggested alternative:
<built-in>: note:   '__builtin_isinf_sign'
node_lib.target.mk:199: recipe for target '/tmp/kibana--node-20180923-22687-1sqmnwq/node-v8.11.4/out/Release/obj.target/node_lib/src/node_crypto.o' failed
make[1]: *** [/tmp/kibana--node-20180923-22687-1sqmnwq/node-v8.11.4/out/Release/obj.target/node_lib/src/node_crypto.o] Error 1
make[1]: *** Waiting for unfinished jobs....
rm 2e172db38e9717471c4feba3120f5ccd28003d5a.intermediate
Makefile:88: recipe for target 'node' failed
make: *** [node] Error 2

READ THIS: https://github.com/Linuxbrew/brew/wiki/troubleshooting
Please do not report this issue to Homebrew/brew or Homebrew/core,
which support macOS only."

```
- What you expected to happen
Kibana will be installed
- Step-by-step reproduction instructions (by running `brew install` commands)
```
brew install kibana
```

I don't have sudo/root privileges on this server and cannot install Linuxbrew to `/home/linuxbrew`. And because of this I cannot fix warnings reported by `brew doctor`. But they looks unrelated.

Information about installed `node`:
```
$ brew info node
node: stable 10.11.0, HEAD
Platform built on V8 to build network applications
https://nodejs.org/
/usr/home/lynpub/.linuxbrew/Cellar/node/10.11.0 (3,939 files, 51MB) *
  Built from source on 2018-09-22 at 16:56:18
From: https://github.com/Linuxbrew/homebrew-core/blob/master/Formula/node.rb
==> Dependencies
Build: pkg-config ✔, python@2 ✔
Recommended: icu4c ✔
Optional: openssl@1.1 ✘
==> Options
--with-openssl@1.1
        Build against Homebrew's OpenSSL instead of the bundled OpenSSL
--without-icu4c
        Build with small-icu (English only) instead of system-icu (all locales)
--without-npm
        npm will not be installed
--HEAD
        Install HEAD version
==> Caveats
Bash completion has been installed to:
  /usr/home/lynpub/.linuxbrew/etc/bash_completion.d
==> Analytics
install: 287,817 (30d), 1,048,403 (90d), 3,626,689 (365d)
install_on_request: 214,464 (30d), 721,893 (90d), 2,244,411 (365d)
build_error: 185 (30d)
```
---
I've found [this](https://github.com/nodejs/node/pull/19196) PR in Node JS repo. Looks like `node` v8.11.4 cannot be compiled with GCC 5.5.0. They fixed this bug in [v8.12.0](https://github.com/nodejs/node/releases/tag/v8.12.0)
> [5a3dc37bc8] - crypto: Use math.h definitions of isnan and isinf (Jeroen Roovers) #19196

**Comment by StudioMaX** [comment]
Uh... Kibana v6.4.1 doesn't support Node JS 8.12.0 (https://github.com/elastic/kibana/blob/v6.4.1/.node-version)
Tried to update Node JS in `kibana.rb` manually and this error was thrown
```
$ brew install kibana
==> Cloning https://github.com/elastic/kibana.git
Updating /usr/home/lynpub/.cache/Homebrew/kibana--git
==> Checking out tag v6.4.1
HEAD is now at 2872c74... eui 3.0.6, fix for ie11 tooltip (#23006)
HEAD is now at 2872c74 eui 3.0.6, fix for ie11 tooltip (#23006)
==> Downloading https://nodejs.org/dist/v8.12.0/node-v8.12.0.tar.xz
######################################################################## 100.0%
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
tar: Ignoring unknown extended header keyword 'SCHILY.dev'
tar: Ignoring unknown extended header keyword 'SCHILY.ino'
tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
==> ./configure --prefix=/usr/home/lynpub/.linuxbrew/Cellar/kibana/6.4.1/libexec/node
==> make install
==> Downloading https://yarnpkg.com/downloads/1.9.4/yarn-v1.9.4.tar.gz
Already downloaded: /usr/home/lynpub/.cache/Homebrew/downloads/8369cb940fbf3440133e8162eeeeb7dfa25b0529e8a789f31188fba42545be78--yarn-v1.9.4.tar.gz
==> yarn kbn bootstrap
Last 15 lines from /usr/home/lynpub/.cache/Homebrew/Logs/kibana/03.yarn:
2018-09-24 08:46:01 -0400

yarn
kbn
bootstrap

yarn run v1.9.4
$ node scripts/kbn bootstrap
Kibana does not support the current Node.js version v8.12.0. Please use Node.js v8.11.4.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

READ THIS: https://github.com/Linuxbrew/brew/wiki/troubleshooting
Please do not report this issue to Homebrew/brew or Homebrew/core,
which support macOS only."

These open issues may also help:
Cannot install Kibana: required "node" v8.11.4 cannot be compiled with GCC 5.5.0 https://github.com/Linuxbrew/homebrew-core/issues/9530
```

I think it's possible to apply [this](https://github.com/nodejs/node/pull/19196/files) patch to Node v8.11.4 at the compilation time, but I don't tried this yet.

**Comment by StudioMaX** [comment]
Fixed by https://github.com/Linuxbrew/homebrew-core/pull/9535
